extension/src/ALObject/ALControl.ts

```diff
@@ -0,0 +1,265 @@
+import * as Common from '../Common';
+import { TransUnit } from "../XLIFFDocument";
+import { ALElement } from "./ALElement";
+import { ALObject } from "./ALObject";
+import { ALPagePart } from "./ALPagePart";
+import { ALProperty } from "./ALProperty";
+import { ALControlType, ALObjectType, ALPropertyType, MultiLanguageType, XliffTokenType } from "./Enums";
+import { MultiLanguageObject } from "./MultiLanguageObject";
+import { XliffIdToken } from "./XliffIdToken";
+
+export class ALControl extends ALElement {
+    type: ALControlType = ALControlType.None;
+    _name?: string;
+    private _value?: string;
+    xliffTokenType: XliffTokenType = XliffTokenType.InheritFromControl;
+    multiLanguageObjects: MultiLanguageObject[] = new Array();
+    controls: ALControl[] = new Array();
+    properties: ALProperty[] = new Array();
+    isALCode: boolean = false;
+
+    constructor(type: ALControlType, name?: string, value?: string) {
+        super();
+        this.type = type;
+        if (name) {
+            this.name = name;
+        }
+        if (value) {
+            this.value = value;
+        }
+    }
+
+    public get name(): string {
+        if (!this._name) {
+            return '';
+        }
+        return this._name;
+    }
+    public set name(name: string) {
+        name = name.trim();
+        if (name.toLowerCase().startsWith('rec.')) {
+            name = name.substr(4);
+        }
+        this._name = Common.TrimAndRemoveQuotes(name);
+    }
+
+    public get value(): string {
+        if (!this._value) {
+            return '';
+        }
+        return this._value;
+    }
+    public set value(value: string) {
+        value = value.trim();
+        if (value.toLowerCase().startsWith('rec.')) {
+            value = value.substr(4);
+        }
+        this._value = Common.TrimAndRemoveQuotes(value);
+    }
+
+
+    public get caption(): string {
+        let prop = this.multiLanguageObjects.filter(x => x.name === MultiLanguageType[MultiLanguageType.Caption])[0];
+        if (!prop) {
```

Comment:
switching this to the true condition would enable an early exit and less spaghetti below